preload/preload-connect-to-doc.html

```diff
@@ -0,0 +1,60 @@
+<!doctype html>
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/preload/resources/preload_helper.js"></script>
+<body>
+<script>
+
+promise_test(async t => {
+    const href = '/common/square.png';
+    const sequence = [];
+    const loaded = new Promise(resolveLoad => {
+        customElements.define("with-preload", class extends HTMLElement {
+            constructor() {
+              super();
+              const shadow = this.attachShadow({ mode: "closed" });
+              const link = document.createElement('link');
+              link.rel = 'preload';
+              link.as = 'image';
+              link.href = href;
+              shadow.appendChild(link);
+              sequence.push('constructed');
+              link.addEventListener('load', () => {
+                  sequence.push('loaded');
+                  resolveLoad();
+              });
+            }
+
+            connectedCallback() {
+                sequence.push('connected');
+            }
+        });
+    });
+
+    const wrapper = document.createElement('with-preload');
+    const timeout = 500;
+    await new Promise(resolve => t.step_timeout(resolve, timeout));
+    document.body.appendChild(wrapper);
+    await loaded;
+    assert_array_equals(sequence, ['constructed', 'connected', 'loaded']);
+}, 'preload link should fetch when attached to a shadow DOM');
```

Comment:
Can you add a similar test without connecting the shadow tree to see that it's not loading?